etcd3/client.py

```diff
@@ -126,6 +174,131 @@ def _get_secure_creds(self, ca_cert, cert_key, cert_cert):
                         cert_cert_file.read()
                     )
 
+    def _discover(self, srv_name):
```

Comment:
It'd make it clearer if you switch this method to be a `@staticmethod`. Just pass `uses_secure_channel` as an arg.